internal/gps/solver.go

```diff
@@ -533,6 +533,49 @@ func (s *solver) solve() (map[atom]map[string]struct{}, error) {
 	return projs, nil
 }
 
+// DeductionErrs maps package import path to errors occurring during deduction.
+type DeductionErrs map[string]error
+
+func (e DeductionErrs) Error() string {
+	return "could not deduce external imports' project roots"
+}
+
+// ValidateParams validates the solver parameters to ensure solving can be completed.
+func ValidateParams(params SolveParameters, sm SourceManager) error {
```

Comment:
Need just a basic test for this - feed it some `SolveParameters` with both valid and invalid import paths.